src/comm/comm_session.hpp

```diff
@@ -17,7 +17,7 @@ enum SESSION_FLAG
 {
     USER_CHALLENGE_ISSUED,
     USER_AUTHED,
-    PEERID_RESOLVED
+    PEERCHALLENGE_RESOLVED
```

Comment:
See the [this comment](https://github.com/HotPocketDev/core/pull/93/files#r410878362) also. We can simplify this enum from 4 values to 2 values. So both user and peer connections would use same two statuses `CHALLENGE_ISSUED` and `CHALLENGE_VERIFIED`. Simplifying further, we can get rid of `session_flags` bitset and introduce enum variable `challenge_status` into `comm_session` class. `session_flags` was originally introduced to keep various custom flags that the socket_session class didn't know about. Now because the challenge concept is common for both user and peer session we do not need custom flags anymore.

Reply:
Change added